c++/src/kj/common.h

```diff
@@ -25,10 +25,31 @@
 
 #pragma once
 
-#if defined(__GNUC__) && !KJ_HEADER_WARNINGS
-#pragma GCC system_header
+#ifdef __GNUC__
+#define KJ_SUPPRESS_WARNINGS _Pragma("GCC system_header")
```

Comment:
Seems like the name `KJ_SUPPRESS_WARNINGS` doesn't really describe what the macro does in the GCC case. Maybe `KJ_SUPPRESS_HEADER_WARNINGS`?

Reply:
Yeah, I struggled this one because if you make it HEADER_WARNINGS it's not accurately describing MSVC is doing. Thoughts?

Reply:
How about we name the pair: `KJ_BEGIN_SYSTEM_HEADER`, `KJ_END_SYSTEM_HEADER`

Reply:
Do you have any concerns about confusion with KJ_BEGIN_HEADER which is conditions by KJ_HEADER_WARNINGS?

Reply:
Nope, I'm fine with those as-is.